data/data-pipeline/data_pipeline/etl/score/etl_score.py

```diff
@@ -286,11 +287,43 @@ def _add_percentiles_to_df(
         something like "3rd grade reading proficiency" and `output_column_name_root`
         may be something like "Low 3rd grade reading proficiency".
         """
-        # Create the "basic" percentile.
-        df[
-            f"{output_column_name_root}"
-            f"{field_names.PERCENTILE_FIELD_SUFFIX}"
-        ] = df[input_column_name].rank(pct=True, ascending=ascending)
+        if (
+            output_column_name_root
+            != field_names.EXPECTED_AGRICULTURE_LOSS_RATE_FIELD
+        ):
+            # Create the "basic" percentile.
+            df[
+                f"{output_column_name_root}"
+                f"{field_names.PERCENTILE_FIELD_SUFFIX}"
+            ] = df[input_column_name].rank(pct=True, ascending=ascending)
+
+        else:
+            # For agricultural loss, we are using whether there is value at all to determine percentile
+            # This is not the most thoughtfully written code, but it works.
+
+            # Take only rows with agrivalue
+            tmp_df = df[df[field_names.AGRICULTURAL_VALUE_BOOL_FIELD] == 1][
+                [input_column_name, field_names.GEOID_TRACT_FIELD]
+            ].copy()
+
+            # Construct a percentile only among those tracts
+            tmp_df["temporary_ranking"] = tmp_df[input_column_name].transform(
+                lambda x: x.rank(pct=True, ascending=True)
+            )
+
+            # Create a map for just those tracts and map it onto the df
+            temporary_ranking = tmp_df.set_index(field_names.GEOID_TRACT_FIELD)[
+                "temporary_ranking"
+            ].to_dict()
+
+            df[
+                f"{output_column_name_root}"
+                f"{field_names.PERCENTILE_FIELD_SUFFIX}"
+            ] = (
+                df[field_names.GEOID_TRACT_FIELD]
+                .map(temporary_ranking)
```

Comment:
lol you raise a very good point.  updating.

Reply:
Okay sorry, so this actually ends up being slower (because you have to join on two columns) or messier (because you have to drop the input column) so for now i am leaving it as a map.